frame/transaction-payment/rpc/runtime-api/src/lib.rs

```diff
@@ -30,5 +32,37 @@ sp_api::decl_runtime_apis! {
 	{
 		fn query_info(uxt: Block::Extrinsic, len: u32) -> RuntimeDispatchInfo<Balance>;
 		fn query_fee_details(uxt: Block::Extrinsic, len: u32) -> FeeDetails<Balance>;
+		fn query_call_info(encoded_call: Bytes) -> Result<RuntimeDispatchInfo<Balance>, DispatchError>;
+		fn query_weight_to_fee(encoded_call: Bytes) -> Result<Balance, DispatchError>;
```

Comment:
What would be the best type to use as a returned error here?
Would be good to have some enum variant or code for the decode error, and be able to bubble the error info from Decode.
Useful for runtime api clients and to map on rpc server level.

Reply:
Why return an error? This is infallible just as the functions you added it next to.